web-common/src/runtime-client/exponential-backoff-tracker.ts

```diff
@@ -1,52 +1,35 @@
-import { asyncWait } from "@rilldata/web-common/lib/waitUtils";
-
 export class ExponentialBackoffTracker {
-  private curTime: number;
   private curRetries = 0;
-  private trackerPeriod: number;
+  private currentDelay: number;
 
   public constructor(
-    private readonly retries: number,
-    /**
-     * Time period within which to trigger the tracker.
-     * Any failure after this will be considered as an intermittent failure.
-     */
-    private readonly trackerTriggerPeriod: number,
-    // time
-    private readonly waitPeriod: number,
+    private readonly maxRetries: number,
+
+    private readonly initialDelay: number,
   ) {
-    this.trackerPeriod = trackerTriggerPeriod;
+    this.currentDelay = initialDelay;
   }
 
   public static createBasicTracker() {
-    return new ExponentialBackoffTracker(5, 1000, 250);
+    return new ExponentialBackoffTracker(5, 1000);
   }
 
-  public async failed(): Promise<boolean> {
-    const lastTime = this.curTime;
-    this.curTime = Date.now();
+  public try = async (fn: () => Promise<void> | void) => {
```

Comment:
Can we keep the older way to defining this class method? While this is valid javascript it is not consistant with other languages.